### PR TITLE
🚧Add render overloads

### LINKED ISSUE #957: 🚧Add render overloads
### Complete The Item Below

- [X] I have updated the title without removing the 🚧 emoji.

### Description

Add new overloads to the `ITextureRenderer` interface.

**The following overloads should be added:**
1. `void Render(ITexture texture, int x, int y, float angle, float size, int layer = 0);`
2. `void Render(ITexture texture, Vector2 pos, float angle, float size, int layer = 0);`
3. `void Render(ITexture texture, int x, int y, float angle, float size, Color color, int layer = 0);`
4. `void Render(ITexture texture, Vector2 pos, float angle, float size, Color color, int layer = 0);`
5. `void Render(IAtlasData atlas, string subTextureName, Vector2 pos, int frameNumber, int layer = 0);`
6. `void Render(IAtlasData atlas, string subTextureName, Vector2 pos, Color color, int frameNumber, int layer = 0);`
7. `void Render(IAtlasData atlas, string subTextureName, Vector2 pos, float angle, int frameNumber, int layer = 0);`
8. `void Render(IAtlasData atlas, string subTextureName, Vector2 pos, float angle, float size, int frameNumber, int layer = 0);`
9. `void Render(IAtlasData atlas, string subTextureName, Vector2 pos, float angle, Color color, int frameNumber, int layer = 0);`
10. `void Render(IAtlasData atlas, string subTextureName, Vector2 pos, float angle, float size, Color color, int frameNumber, int layer = 0);`
11. `void Render(IAtlasData atlas, string subTextureName, Vector2 pos, float angle, float size, Color color, RenderEffects effects, int frameNumber = 0, int layer = 0);`

**Reason:**
The reason for this is to be able to easily render a texture and use size without having to be forced to use the overload with the largest quantity of parameters when all you need is the size.

The Velaptor docs need to be updated by finding any code samples that use more complicated texture renderer methods than necessary.  

### Acceptance Criteria

```[tasklist]
### The items to complete to satisfy the Definition of Done.
- [x] All unit tests executed and passed locally after PR work is complete. _(Why? Refer to 'Additional Information' below.)_
- [x] Issue created to update various docs to use simpler overloads in the code samples.
- [x] `void Render(ITexture texture, int x, int y, float angle, float size, int layer = 0);` as been added
- [x] `void Render(ITexture texture, Vector2 pos, float angle, float size, int layer = 0);` as been added
- [x] `void Render(ITexture texture, int x, int y, float angle, float size, Color color, int layer = 0);` as been added
- [x] `void Render(ITexture texture, Vector2 pos, float angle, float size, Color color, int layer = 0);` as been added
- [x] `void Render(IAtlasData atlas, string subTextureName, Vector2 pos, int frameNumber, int layer = 0);` has been added
- [x] `void Render(IAtlasData atlas, string subTextureName, Vector2 pos, Color color, int frameNumber, int layer = 0);` has been added
- [x] `void Render(IAtlasData atlas, string subTextureName, Vector2 pos, float angle, int frameNumber, int layer = 0);` has been added
- [x] `void Render(IAtlasData atlas, string subTextureName, Vector2 pos, float angle, float size, int frameNumber, int layer = 0);` has been added
- [x] `void Render(IAtlasData atlas, string subTextureName, Vector2 pos, float angle, Color color, int frameNumber, int layer = 0);` has been added
- [x] `void Render(IAtlasData atlas, string subTextureName, Vector2 pos, float angle, float size, Color color, int frameNumber, int layer = 0);` has been added
- [x] 'void Render(IAtlasData atlas, string subTextureName, Vector2 pos, float angle, float size, Color color, RenderEffects effects, int frameNumber = 0, int layer = 0);' has been added
- [x] size and angle parameters swapped for the private `RenderBase` method
```

### ToDo Items

```[tasklist]
### The items to complete to satisfy the Definition of Done.
- [x] Change type labels added to this issue.  Refer to the _**Change Type Labels**_ section below.
- [x] Priority label added to this issue.  Refer to the _**Priority Type Labels**_ section below.
- [x] Issue linked to the correct milestone _(if applicable)_.
```

### Issue Dependencies

_No response_

### Related Work

_No response_

### Additional Information:

**_<details closed><summary>Unit Tests</summary>_**

Reasons for local unit test execution:
- Unit tests might pass locally but not in the CI environment during the status check process or vice-versa.
- Tests might pass on the developer's machine but not necessarily on the code reviewer's machine.
- If you notice that the test status check has passed but the test failed locally, please notify a project maintainer!

💡Warning💡
If the unit tests pass remotely and are not executed locally, this means we could be letting a bug slip into production.
Though bugs will always exist in some capacity, we should all do our part to help prevent them from happening.
</details>

**_<details closed><summary>Change Type Labels</summary>_**

| Change Type              | Label                     |
|:-------------------------|:--------------------------|
| Bug Fixes                | `🐛bug`                   |
| Breaking Changes         | `🧨breaking changes`      |
| New Feature              | `✨new feature`           |
| CICD Changes             | `♻️cicd`                  |
| Config Changes           | `⚙️config`                |
| Performance Improvements | `🏎️performance`           |
| Code Doc Changes         | `🗒️documentation/code`    |
| Product Doc Changes      | `📝documentation/product` |
</details>

**_<details closed><summary>Priority Type Labels</summary>_**

| Priority Type       | Label             |
|---------------------|-------------------|
| Low Priority        | `low priority`    |
| Medium Priority     | `medium priority` |
| High Priority       | `high priority`   |
</details>

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### PR DESCRIPTION
**Pull request by CalvinWilkinson**
### **Description:**

To allow this pull request to be merged, please make sure that the following items of this pull request are complete.

This pull request closes #957

✅The pull request head branch must be a feature branch with the syntax 'feature/<issue-num>-sync-testing'.  
✅The pull request base branch must be a 'main' or 'preview' branch..  
✅The pull request head branch contains a valid issue number.  
✅The pull request title matches the linked issue title exactly.  
✅The pull request assignees match the assignees of the issue.  
✅The pull request labels match the labels of the issue.  
✅The pull request organizational projects match the organizational projects of the issue.  
✅The pull requests milestone matches the milestone of the issue.  

---

### **Additional Info:**
1. Pull requests are automatically synced with the associated issue upon creation by one of the KinsonDigital workflow bots. 
2. The associated issue is the issue number that is embedded in the pull request head branch.
3. The list above will be automatically updated as the pull request's various settings match or do not match the associated issue.
4. To manually sync the pull request to the issue, create a comment with the `[run-sync]` command.
	> **Note** You must be an admin member of the organization to use this command.

---

- [x] Sync with the issue.  Use this to enable or disable this pull request from syncing with its associated issue.

### CODE REVIEW

File: Velaptor/Graphics/Renderers/ITextureRenderer.cs [hunk 1]
```diff
@@ -96,8 +97,100 @@ public interface ITextureRenderer
     ///         <item>Texture 5</item>
     ///     </list>
     /// </remarks>
+    [SuppressMessage("ReSharper", "UnusedMemberInSuper.Global", Justification = "Public API for users.")]
     void Render(ITexture texture, int x, int y, float angle, int layer = 0);
 
+    /// <summary>
+    /// Renders the given texture at the given <paramref name="x"/> and <paramref name="y"/> coordinates <paramref name="angle"/>,
```
Comment (KinsonDigitalAdmin) [inline Velaptor/Graphics/Renderers/ITextureRenderer.cs]: Add a comma after the word `coordinates`

```suggestion
    /// Renders the given texture at the given <paramref name="x"/> and <paramref name="y"/> coordinates, <paramref name="angle"/>,
```

File: Velaptor/Graphics/Renderers/ITextureRenderer.cs [hunk 2]
```diff
@@ -96,8 +97,100 @@ public interface ITextureRenderer
...
+    /// and <paramref name="size"/>.
+    /// </summary>
+    /// <param name="texture">The texture to render.</param>
+    /// <param name="x">The X location of the texture.</param>
+    /// <param name="y">The Y location of the texture.</param>
+    /// <param name="angle">The angle of rotation in degrees of the rendering.</param>
```
Comment (KinsonDigitalAdmin) [inline Velaptor/Graphics/Renderers/ITextureRenderer.cs]: Remove the text `of the rendering.`

Do this for all copies of this param doc.

```suggestion
    /// <param name="angle">The angle of rotation in degrees.</param>
```

File: Velaptor/Graphics/Renderers/ITextureRenderer.cs [hunk 3]
```diff
@@ -96,8 +97,100 @@ public interface ITextureRenderer
...
+    /// <param name="size">The size to render the texture. 1 is for 100%/normal size.</param>
```
Comment (KinsonDigitalAdmin) [inline Velaptor/Graphics/Renderers/ITextureRenderer.cs]: Refactor this param doc.

Do this for all copies of this param docs.

```suggestion
    /// <param name="size">The size to render the texture. 1 is for 100%, which is the normal size.</param>
```

File: Velaptor/Graphics/Renderers/ITextureRenderer.cs [hunk 4]
```diff
@@ -474,4 +663,294 @@ public interface ITextureRenderer
     ///     </list>
     /// </remarks>
     void Render(ITexture texture, Rectangle srcRect, Rectangle destRect, float size, float angle, Color color, RenderEffects effects, int layer = 0);
+
+    /// <summary>
+    /// Renders the given <see cref="Texture"/> using the given parameters.
```
Comment (KinsonDigitalAdmin) [inline Velaptor/Graphics/Renderers/ITextureRenderer.cs]: Fix all instances of this doc for all render overloads with an atlas texture.

```suggestion
    /// Renders the <see cref="Texture"/> in the given <paramref name="atlasData"/> using the given parameters.
```

File: Velaptor/Graphics/Renderers/TextureRenderer.cs [hunk 5]
```diff
@@ -230,7 +315,262 @@ public void Render(
...
+
+    /// <inheritdoc/>
+    /// <exception cref="ArgumentNullException">Thrown if the <paramref name="atlas"/> is null.</exception>
+    /// <exception cref="InvalidOperationException">
+    ///     Thrown if the <see cref="IBatcher.Begin"/> has not been invoked before rendering.
+    /// </exception>
+    /// <exception cref="ArgumentException">
+    ///     Thrown if the source rectangle width or height is less than or equal to 0.
+    /// </exception>
+    public void Render(IAtlasData atlas, string subTextureName, Vector2 pos, float angle, float size, int frameNumber = 0, int layer = 0)
```
Comment (KinsonDigitalAdmin) [inline Velaptor/Graphics/Renderers/TextureRenderer.cs:443]: Add exception docs to all atlas render overloads that throw an exception due to an invalid frame number.

**Review by KinsonDigitalAdmin — CHANGES_REQUESTED** [review]

**Comment by CalvinWilkinson** [comment]
> [!Note]
> Improved the https://github.com/KinsonDigital/Velaptor/labels/%F0%9F%93%91documentation-code for the `ITextureRenderer` interface

**Review by KinsonDigitalAdmin — APPROVED** [review]

**Comment by CalvinWilkinson** [comment]
[run-sync]

**Comment by CalvinWilkinson** [comment]
> [!Note]
> Added the following https://github.com/KinsonDigital/Velaptor/labels/%E2%9C%A8new-feature 's by creating new render texture overloads.
> - `void Render(ITexture texture, int x, int y, float angle, float size, int layer = 0);`
> - `void Render(ITexture texture, Vector2 pos, float angle, float size, int layer = 0);`
> - `void Render(ITexture texture, int x, int y, float angle, float size, Color color, int layer = 0);`
> - `void Render(ITexture texture, Vector2 pos, float angle, float size, Color color, int layer = 0);`
> - `void Render(IAtlasData atlas, string subTextureName, Vector2 pos, int frameNumber, int layer = 0);`
> - `void Render(IAtlasData atlas, string subTextureName, Vector2 pos, Color color, int frameNumber, int layer = 0);`
> - `void Render(IAtlasData atlas, string subTextureName, Vector2 pos, float angle, int frameNumber, int layer = 0);`
> - `void Render(IAtlasData atlas, string subTextureName, Vector2 pos, float angle, float size, int frameNumber, int layer = 0);`
> - `void Render(IAtlasData atlas, string subTextureName, Vector2 pos, float angle, Color color, int frameNumber, int layer = 0);`
> - `void Render(IAtlasData atlas, string subTextureName, Vector2 pos, float angle, float size, Color color, int frameNumber, int layer = 0);`
> - `void Render(IAtlasData atlas, string subTextureName, Vector2 pos, float angle, float size, Color color, RenderEffects effects, int frameNumber = 0, int layer = 0);`

**Pull request merged**